core/src/main/java/org/elasticsearch/index/query/FieldMaskingSpanQueryParser.java

```diff
@@ -76,9 +76,6 @@ public QueryBuilder fromXContent(QueryParseContext parseContext) throws IOExcept
                 }
             }
         }
-        if (inner == null) {
```

Comment:
inner can still be null here I'm afraid, if we never go in the if branch that parses it, that we had the queryFound flag for in other queries. I think this check is still valid